### ISSUE TITLE
Infinite loop in cureLocalIntersections

### ISSUE THREAD

**Issue by randomdude7332** [mapbox/earcut.hpp#89]
The issue appears to be fairly simple (incorrect linked list pointers due to out of order items removal), but I'm not familiar with the code, so it might be something more serious. Patched version seem to produce correct-looking output, at least.

Diff (a PR seems rather excessive): [diff.txt](https://github.com/mapbox/earcut.hpp/files/5789936/diff.txt)
And, one of the offending polygons: [polygon.txt](https://github.com/mapbox/earcut.hpp/files/5789934/polygon.txt)

**Comment by mourner** [comment]
> a PR seems rather excessive

Not at all! The biggest advantage is that it will run all the tests against the fix.

**Comment by randomdude7332** [comment]
Apologies for the delay, I don't use github so it took a while to notice. Opened a PR.

**Comment by mourner** [comment]
Appears to be fixed in v3.2.3
